htdocs/core/class/commonobject.class.php

```diff
@@ -2511,6 +2511,27 @@ function update_note($note,$suffix='')
 				$this->note = $note;      // deprecated
 				$this->note_private = $note;
 			}
+			if($suffix != '_private' && in_array($this->table_element, array('commande_fournisseur', 'commande', 'propal', 'facture_fourn', 'facture' ))){
```

Comment:
The "update_note" method is a CRUD method to affect the database values. I would prefer to see code to build PDF that is not related with DAO principle after the call of update_note